pkg/nodeconfig/payload/payload.go

```diff
@@ -165,8 +198,14 @@ $cni_template=@'
 }
 '@
 
+$hns_network = Retry-GetHnsNetwork
+# If the HNS network is never found, quit.
+if ($null -eq $hns_network) {
+  Write-Host "HNS_NETWORK not found after retries."
+  exit 1
```

Comment:
Seems like [throw](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_throw?view=powershell-7.5) should be used rather than Write-Host + exit